string cmd
float32[] args